source/common/upstream/upstream_impl.h

```diff
@@ -101,6 +102,80 @@ class UpstreamLocalAddressSelectorImpl : public UpstreamLocalAddressSelector {
   std::vector<UpstreamLocalAddress> upstream_local_addresses_;
 };
 
+/**
+ * Class for LBPolicies
+ * Uses a absl::variant to store pointers for the LBPolicy
+ */
+class LBPolicyConfig {
+public:
+  LBPolicyConfig(const envoy::config::cluster::v3::Cluster& config);
+
+  OptRef<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig> lbRoundRobinConfig() const {
+    if (auto lb_ptr = absl::get_if<
+            std::unique_ptr<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>>(
+            &lbPolicy_)) {
+      return makeOptRefFromPtr<const envoy::config::cluster::v3::Cluster::RoundRobinLbConfig>(
+          (*lb_ptr).get());
```

Comment:
OOC can the `makeOptRefFromPtr` be avoided here?
I see that in the original code `return *lb_round_robin_config_;` was used, so the question if returning `*lb_ptr` is possible in this case?

Reply:
This makes sense, I applied this change into the new helper function in the class.